webpack.config.js

```diff
@@ -0,0 +1,115 @@
+const webpack = require('webpack');
+const path = require('path');
+const ManifestPlugin = require('webpack-manifest-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const CopyPlugin = require('copy-webpack-plugin');
+const exec = require('child_process').exec;
+const config = require('./gulp/config');
+
+const outputPath = path.resolve(`${config.outputPath}/${config.jsDirectory}`);
+const jsSource = `./${config.sourcePath}/${config.jsDirectory}`;
+const cleanPaths = [
+  `${config.outputPath}/fonts/*`,
+  `${config.outputPath}/images/*`,
+  `${config.outputPath}/stylesheets/*`,
+  `${config.outputPath}/javascripts/*`
+];
+const entries = {
+  main: [`${jsSource}/main.js`],
+  raven: [`${jsSource}/raven.js`],
+  styleguide: [`${jsSource}/styleguide/styleguide.jsx`],
+  survey: [`${jsSource}/surveys/survey.js`],
+  survey_admin: [`${jsSource}/surveys/survey-admin.js`],
+  survey_results: [`${jsSource}/surveys/survey-results.jsx`],
+  campaigns: [`${jsSource}/campaigns.js`],
+  charts: [`${jsSource}/charts.js`],
+  tinymce: [`${jsSource}/tinymce.js`],
+  embed_course_stats: [`${jsSource}/embed_course_stats.js`]
+};
+
+const wpPlugins = [];
+const absoluteOutputPath = path.join(process.cwd(), config.outputPath);
+const jqueryUlsPath = './node_modules/jquery/dist/jquery.min.js';
+const mode = 'production';
+
+wpPlugins.push(new CleanWebpackPlugin({
+  cleanOnceBeforeBuildPatterns: cleanPaths.map(relative => path.join(process.cwd(), relative))
+}));
+
+wpPlugins.push({
```

Comment:
Can you add some comments to explain the strategy here for the I18n pipeline?

Also, is there a simple way to add fingerprinting? A common problem is new interface messages get deployed and users get the new main.js build that uses them, but they see 'message missing' until they clear their cached I18n files since they don't get fingerprinted.

Reply:
Sure. Though I don't think i18n is part of the main bundle.